app/models/gem_dependent.rb

```diff
@@ -0,0 +1,56 @@
+class GemDependent
+  attr_reader :gem_names
+
+  def initialize(gem_names)
+    @gem_information = {}
+    @gem_names = gem_names
+  end
+
+  def fetch_dependencies
+    dependencies = []
+    @gem_names.each { |g| @gem_information[g] = "deps/v1/#{g}" }
+
+    @gem_information.each do |gem_name, cache_key|
+      if (dependency = memcached_gem_info[cache_key])
+        # Fetch the gem's dependencies from the cache
+        StatsD.increment 'gem_dependent.memcached.hit'
+        dependencies << dependency
+      else
+        # Fetch the gem's dependencies from the database
+        StatsD.increment 'gem_dependent.memcached.miss'
+        result = fetch_dependency_from_db(gem_name)
```

Comment:
Do you mean to update the cache? We do that on the next line with `memcached_client.write`.